spec/lib/services/migration/failure_manager_spec.rb

```diff
@@ -0,0 +1,70 @@
+require "rails_helper"
+
+RSpec.describe Migration::FailureManager, in_memory_rails_cache: true do
+  let(:instance) { described_class.new(data_migration:) }
+
+  describe "#record_failures" do
+    subject { instance.record_failure(item, failure_message) }
+
+    context "with correct params" do
+      let(:data_migration) { create(:data_migration, model: :statement, failure_count: 2) }
+      let(:item) { create(:ecf_migration_statement) }
+      let(:failure_message) { "Test failure" }
+
+      it "records a failure" do
+        subject
+
+        failure = Rails.cache.read("migration_failure_#{data_migration.model}_#{data_migration.id}_1")
+
+        expect(failure[:model]).to eq("statement")
+        expect(failure[:id]).to eq(item.id)
+        expect(failure[:failure_message]).to eq(failure_message)
+      end
+    end
+
+    context "with incorrect params" do
+      let(:data_migration) { create(:data_migration) }
+      let(:item) { "" }
+      let(:failure_message) { "Test failure" }
+
+      it "returns nil" do
+        expect(subject).to be_nil
```

Comment:
This is testing the return value of the method, not that it hasn't recorded a failure?

```
expect(instance.all_failures).to be_empty
```